lib/symlink/index.js

```diff
@@ -51,22 +51,24 @@ function symlink(outFolder, opt) {
       srcPath = path.relative(file.base, srcPath);
     }
 
-    // TODO: make DRY with .dest()
-    var dirMode = number(opt.dirMode, file);
-    var writeFolder = path.dirname(file.path);
-
-    fo.mkdirp(writeFolder, dirMode, onMkdirp);
+    // Because fs.symlink does not allow atomic overwrite option with flags, we
+    // delete and recreate if the link already exists and overwrite is true.
+    if (file.flag === 'w') {
+      // TODO What happens when we call unlink with windows junctions?
+      fs.unlink(file.path, onUnlink);
+    } else {
+      fs.symlink(srcPath, file.path, symType, onSymlink);
+    }
 
-    function onMkdirp(mkdirpErr) {
-      if (mkdirpErr) {
-        return callback(mkdirpErr);
+    function onUnlink(unlinkErr) {
+      if (unlinkErr && unlinkErr.code !== 'ENOENT') {
```

Comment:
Even though it's not being used another place, I think it would be clearer to someone reading the code in the future (and me, when I read it the first time).